mozjs-sys/build.rs

```diff
@@ -889,3 +911,35 @@ fn link_static_lib_binaries(build_dir: &Path) -> Result<(), std::io::Error> {
     println!("cargo:rustc-link-lib=static=jsglue");
     Ok(())
 }
+
+/// Returns the value `cc-rs` would use for `var_base`
+///
+/// Since we build parts of our code without cc-rs by directly invoking spidermonkey,
+/// we should first adjust key environment variables like `CC`, `CXX`, `AR` etc. to
+/// have the values that users of `cc-rs` would expect.
+///
+/// Adapted from https://github.com/rust-lang/cc-rs/blob/3ba23569a623074748a3030f382afd22483555df/src/lib.rs#L3617
+fn get_cc_rs_env(var_base: &str) -> Option<String> {
+    get_cc_rs_env_os(var_base).map(|val| val.to_str().expect("Not a valid string.").to_string())
+}
+
+/// Like `get_cc_rs_env()` but returns the OsString value.
+fn get_cc_rs_env_os(var_base: &str) -> Option<OsString> {
+    fn get_env(var: &str) -> Option<OsString> {
+        let value = env::var_os(var)?;
+        println!("cargo:rerun-if-env-changed={}", var);
+        Some(value)
+    }
```

Comment:
I am not sure about this, I would run this unconditionally just to be on the safe side.

Reply:
Thanks for catching this. Indeed, we want to recompile if a previously unset variable is now set.